Source/DataSources/RectangleGeometryUpdater.js

```diff
@@ -527,16 +541,28 @@ define([
         }
         //>>includeEnd('debug');
 
+        var geometryUpdater = this._geometryUpdater;
+        var onTerrain = geometryUpdater._onTerrain;
+        if (defined(this._primitive) && geometryUpdater._isConstant) {
+            return;
+        }
+
         var primitives = this._primitives;
-        primitives.removeAndDestroy(this._primitive);
+        var groundPrimitives = this._groundPrimitives;
+        if (onTerrain) {
+            groundPrimitives.removeAndDestroy(this._primitive);
+        }
+        else {
+            primitives.removeAndDestroy(this._primitive);
+        }
         primitives.removeAndDestroy(this._outlinePrimitive);
         this._primitive = undefined;
         this._outlinePrimitive = undefined;
 
-        var geometryUpdater = this._geometryUpdater;
         var entity = geometryUpdater._entity;
         var rectangle = entity.rectangle;
-        if (!entity.isShowing || !entity.isAvailable(time) || !Property.getValueOrDefault(rectangle.show, time, true)) {
+        var isAvailable = entity.isAvailable(time);
```

Comment:
Not sure. I think it had something to do with changing an older version 6 months ago and then merging in current master. I don't recall changing it.